Tests/Runtime/Reporter/ReportUploadGuardServiceTests.cs

```diff
@@ -0,0 +1,82 @@
+using BugSplatUnity.Runtime.Reporter;
+using BugSplatUnity.Runtime.Settings;
+using NUnit.Framework;
+using System;
+using UnityEngine;
+
+namespace BugSplatUnity.RuntimeTests.Reporter
+{
+    public class ReportUploadGuardServiceTests
+    {
```

Comment:
Really well done here too! I like that you covered all of the cases. Here's a cool tip, if you find yourself in a similar scenario in the future, but there are a ton more cases, you can use `Parameterized Tests` to pass arguments to your tests 🙂.

Here's a [Gist](https://gist.github.com/muditlambda/37b9b7abdf2692803b912a0bc2aea642#file-testcase-cs-L19) I found for reference.

Reply:
That's a sweet test feature, I'll keep that in mind if I run into in going into the future!